Extend rising bubble example (#1558)
lethe-cfd/lethe: doc/source/examples/multiphysics/rising-bubble/rising-bubble.rst

Case 2
~~~~~~~

The same python post-processing code (``rising-bubble.py``) is used for test case 2, with ``-c 2`` instead.

.. code-block:: text
  :class: copy-button

  python3 ./rising-bubble.py -f rising-bubble-sharp -c 2
  
The following image shows the shape of the bubble after :math:`3` seconds of simulation, and compares it with results obtained by three different codes reported in the work of Hysing *et al.* [#hysing2009]_: TP2D, FreeLIFE and MooNMD.

.. image:: images/sharp-bubble-contour-case2.png
    :alt: bubble
    :align: center
    :width: 500

The barycenter position and velocity of the bubble are also compared to the results from [#hysing2009]_. The following figures show good agreement with the reference.

.. image:: images/sharp-ymean-t-case2.png
    :alt: ymean_t
    :align: center
    :width: 500

.. image:: images/sharp-bubble-rise-velocity-case2.png
    :alt: bubble_rise_velocity
    :align: center
    :width: 500

.. _Interface Regularization Methods Comparison:

---------------------------------------------
Interface Regularization Methods Comparison
---------------------------------------------

Parameter Files
~~~~~~~~~~~~~~~~

> For the methods other than ``projection-based interface sharpening``, the ``.prm`` file is modified as follows. In the ``VOF`` subsection, the ``interface regularization method`` is changed to ``geometric interface reinitialization`` or ``algebraic interface reinitialization``. The associated parameter files, ``rising-bubble-geo.prm`` and ``rising-bubble-alge.prm`` respectively, are avaible in the example's folder. The subsections are modified according to each regularization method: 

Reviewer: ```suggestion
For the methods other than ``projection-based interface sharpening``, the ``.prm`` file is modified as follows. In the ``VOF`` subsection, the ``interface regularization method`` is changed to ``geometric interface reinitialization`` or ``algebraic interface reinitialization``. The associated parameter files, ``rising-bubble-geo.prm`` and ``rising-bubble-alge.prm`` respectively, are available in the example's folder. The subsections are modified according to each regularization method: 
```

* With the geometric method, the phase fraction field is regularized using the signed distance from the interface, as described in :doc:`../../../theory/multiphase/cfd/vof` theory guide. The ``max reinitialization distance`` parameter is set to :math:`0.032` and we select the ``piecewise polynomial`` function to transform the signed distance in a phase fraction field.

.. code-block:: text

    subsection interface regularization method
      set type       = geometric interface reinitialization
      set frequency  = 20
      set verbosity  = verbose
      subsection geometric interface reinitialization
        set max reinitialization distance = 0.032
        set transformation type           = piecewise polynomial
      end
    end

> * For the algebraic method, a intermidiary PDE is solved to compress the interface until reaching a pseudo-steady-state. This PDE is described in :doc:`../../../theory/multiphase/cfd/vof` theory guide. Setting the ``steady-state criterion`` to :math:`10^{-3}` yielded good results.

Reviewer: ```suggestion
* For the algebraic method, an intermediary PDE is solved to compress the interface until reaching a pseudo-steady-state. This PDE is described in :doc:`../../../theory/multiphase/cfd/vof` theory guide. Setting the ``steady-state criterion`` to :math:`10^{-3}` yielded good results.
```

.. code-block:: text

    subsection interface regularization method
      set type      = algebraic interface reinitialization
      set frequency = 20
      set verbosity = verbose
      subsection algebraic interface reinitialization
        set steady-state criterion = 1e-3
      end
    end

Running the Simulations
~~~~~~~~~~~~~~~~~~~~~~~

To run the simulations for the geometric and algebraic regularization methods:

.. code-block:: text
  :class: copy-button

  mpirun -np 8 lethe-fluid rising-bubble-geo.prm

.. code-block:: text
  :class: copy-button

  mpirun -np 8 lethe-fluid rising-bubble-alge.prm

> We are interested in four metrics for this comparison: the barycenter position and velocity, the bubble shape, and the volume conservation. To compare these metrics between the three regularization methods, the python post-processing script ``rising-bubble-comparison.py`` is used:

Reviewer: Should we try to merge the two post-processing files ? From the outside they look a bit redundant I think

.. code-block:: text
  :class: copy-button

>   python3 ./rising-bubble.py -s rising-bubble-sharp -g rising-bubble-geo -a rising-bubble-alge -c 1

Reviewer: ```suggestion
  python3 ./rising-bubble-comparison.py -s rising-bubble-sharp -g rising-bubble-geo -a rising-bubble-alge -c 1
```
